synapseclient/__init__.py

```diff
@@ -266,28 +266,44 @@
 
 """
 
+from .__version__ import __version__
+
+# public APIs
 from .client import Synapse, login
 from .client import PUBLIC, AUTHENTICATED_USERS
 from .client import ROOT_ENTITY
 
 from .activity import Activity
 from .entity import Entity, Project, Folder, File, Link, DockerRepository
 from .evaluation import Evaluation, Submission, SubmissionStatus
-from .table import Schema, EntityViewSchema, Column, RowSet, Row, as_table_columns, Table, PartialRowset, EntityViewType
+from .table import Schema, EntityViewSchema, Column, RowSet, Row, as_table_columns, Table, PartialRowset,\
+    EntityViewType, build_table
 from .team import Team, UserProfile, UserGroupHeader, TeamMember
 from .wiki import Wiki
 
+
 from synapseclient.core.version_check import check_for_updates
 from synapseclient.core.version_check import release_notes
+from synapseclient.core.logging_setup import DEBUG_LOGGER_NAME, DEFAULT_LOGGER_NAME
+
 
-import json
-import pkg_resources
-__version__ = json.loads(pkg_resources.resource_string('synapseclient',
-                                                       'synapsePythonClient').decode())['latestVersion']
+__all__ = [
+    # objects
+    'Synapse', 'Activity', 'Entity', 'Project', 'Folder', 'File', 'Link', 'DockerRepository', 'Evaluation',
+    'Submission', 'SubmissionStatus', 'Schema', 'EntityViewSchema', 'Column', 'Row', 'RowSet', 'Table', 'PartialRowset',
+    'Team', 'UserProfile', 'UserGroupHeader', 'TeamMember', 'Wiki',
+    # functions
+    'login', 'build_table', 'as_table_columns', 'check_for_updates', 'release_notes',
+    # enum
+    'EntityViewType',
+    # constants
+    'PUBLIC', 'AUTHENTICATED_USERS', 'ROOT_ENTITY', 'DEBUG_LOGGER_NAME', 'DEFAULT_LOGGER_NAME']
 
+
+# ensure user-agent is set to track Synapse Python client usage
 import requests
 USER_AGENT = {'User-Agent': 'synapseclient/%s %s' % (__version__, requests.utils.default_user_agent())}
 
-from synapseclient.core import logging_setup as __logingsetup
```

Comment:
We import logging setup to monkey-patch logging behavior. Perhaps there is a better way to do this, but it should not be removed altogether